src/libs/routes/validationHandler.ts

```diff
@@ -1,54 +1,60 @@
+        const inObject = config[keys];
+        inObject.in.forEach(inside => {
+            let value = req[inside][keys];
+            const a = {
+                key : '',
+                location: '',
+                errorMessage: ''
             };
-            errors.push(err);
-        }
-        if ((!keys.required) && !(request)) {
-            return request = keys.default;
-        }
-        if (
-            (((keys.number) && !(Number.isInteger(Number(request)))) ||
-                ((keys.string) && !(typeof request === 'string')))
-        ) {
-            const err = {
-                key: `${key}`,
-                location: `${keys.in}`,
-                errorMessage: `${keys.errorMessage || 'incorrect Type'}`
-            };
-            errors.push(err);
-        }
-        if ((keys.isObject) && !(typeof (request) === 'object')) {
-            const err = {
-                key: `${key}`,
-                location: `${keys.in}`,
-                errorMessage: `${keys.errorMessage || 'not an Object'}`
-            };
-            errors.push(err);
-        }
-        if ((regex) && (!regex.test(request))) {
-            const err = {
-                key: `${key}`,
-                location: `${keys.in}`,
-                errorMessage: `${request} is not valid`
-            };
-            errors.push(err);
-        }
+            if ((inObject.required) && !(value)) {
+                a.key = keys;
+                a.location = inside;
+                a.errorMessage = inObject.errorMessage || `${keys} is required`;
+                error.push(a);
+                return;
+            }
+            value = value || inObject.default;
+            if (!value) {
+                return;
+            }
+            if ((inObject.number) && !(Number.isInteger(Number(value)))) {
+                a.key = keys;
+                a.location = inside;
+                a.errorMessage = inObject.errorMessage || `${keys}'s type is not number`;
+                error.push(a);
+                return;
+            }
+            if ((inObject.string) && !(typeof value === 'string')) {
+                a.key = keys;
+                a.location = inside;
+                a.errorMessage = inObject.errorMessage || `${keys}'s type is not string`;
+                error.push(a);
+                return;
+            }
+            const regex = inObject.regex;
+            if ((regex) && !regex.test(value)) {
+                a.key = keys;
+                a.location = inside;
+                a.errorMessage = inObject.errorMessage || `${keys} is invalid`;
+                error.push(a);
+                return;
+            }
+            if (inObject.isObject && (!(typeof value === 'object') || !(Object.entries(value).length))) {
+                a.key = keys;
+                a.location = inside;
+                a.errorMessage = `${keys} is invalid`;
+                error.push(a);
+                return;
+            }
```

Comment:
do not object name as 'a, b, c' instead use temp or some well-defined vairable

Reply:
Used temp variable